### PR TITLE
Kaleem neslit.9501.merge all oveview

### PR DESCRIPTION
**Pull request by KaleemNeslit**
## Link to Issue
Closes: #9501 

## Description of Changes
- Added the Tabs in discussion page to show "All and Overview " page 
- keep the  functionality   for admins to set the default view of the community.

## "How We Fixed It"

- Added the Tabs in discussion page to show "All and Overview " page 
- keep the  functionality   for admins to set the default view of the community.
- 

## Test Plan
Goto and community and check the tabs to navigate btw all and overView page
-go community and then  in Admin Capabilities sidebar click the Community Profile and then scroll to Default page and then select the page type as default.
Here is the video of  test 

https://github.com/user-attachments/assets/240e16a7-9971-4e27-b943-fffed2653d80

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/overview/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/overview/index.tsx:1]: this gets double padded in discussions page

image

File: packages/commonwealth/client/scripts/views/pages/overview/index.tsx [hunk 1]
```diff
@@ -18,8 +16,16 @@ import { TopicSummaryRow } from './TopicSummaryRow';
 
 const OverviewPage = () => {
   const navigate = useCommonNavigate();
-  const { isWindowSmallInclusive } = useBrowserWindow({});
-  const user = useUserStore();
+  const topicNameFromURL = splitAndDecodeURL(location.pathname);
+  useEffect(() => {
+    if (topicNameFromURL === 'overview') {
+      const params = new URLSearchParams();
+      params.set('tab', 'overview');
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/overview/index.tsx:23]: you'd have to update `HeaderWithFilters.tsx -> onFilterSelect()` logic for URL params setting to work correctly with this section. ex: `?featured=oldest or anything after '?'` should get removed when switching from 'all' to 'overview' tab.

File: packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx [hunk 2]
```diff
@@ -199,13 +216,26 @@ export const HeaderWithFilters = ({
...
+          </CWTabsRow>
+        ) : (
+          <CWText type="h3" fontWeight="semiBold" className="header-text">
+            {isUndefined(topic)
+              ? isOnArchivePage
+                ? 'Archived'
+                : 'All Discussions'
+              : topic}
+          </CWText>
+        )}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:238]: lets store the active tab in url as we do elsewhere. ex: visiting `/discussions?tab=all` should take user to all discussions and `/discussions?tab=overview` should take to overview + lets make sure the URL updates sync with the filters in discussion section

File: packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:1]: filters are for the threads list api on the discussion section, lets remove them when active tab is overview as they won't work there.

**Comment by dillchen** [comment]
Hey @KaleemNeslit sorry we updated the design -> Please see this file:

https://www.figma.com/design/dGPiswtnIfQIzSNOlB4MNC/Sticky-editor-%2B-Sidebar-%2B-Thread-UI?node-id=789-4056&t=0WhzNPyv8wyiMzgb-4

image

**Review by dillchen — CHANGES_REQUESTED** [review]
please see previous comment

**Comment by KaleemNeslit** [comment]
> please see previous comment

sure i am looking into it

**Comment by KaleemNeslit** [comment]
@dillchen Sorry I didn't get the requirements here, Is it this change of [design](https://github.com/hicommonwealth/commonwealth/issues/9502) you are talking about ?
If it's then i'm already assigned this task and will work on it, If no then can you please elaborate on it what changes are required for UI?

**Comment by dillchen** [comment]
For visibility, we're gonna use these for icons for All / Overview: https://www.figma.com/design/dGPiswtnIfQIzSNOlB4MNC/Sticky-editor-%2B-Sidebar-%2B-Thread-UI?node-id=849-2150&t=0WhzNPyv8wyiMzgb-4

**Comment by KaleemNeslit** [comment]
image
@masvelio @mzparacha  @Israellund  i made changes according to new design . I change the tabs with select

**Review by dillchen — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/pages/overview/index.tsx [hunk 3]
```diff
@@ -30,6 +39,7 @@ const OverviewPage = () => {
     apiEnabled: !!communityId,
   });
 
+  console.log('i am called');
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/overview/index.tsx:42]: please remove this

**Review by mzparacha — APPROVED** [review]
The all/overview dropdown gets removed when any filter is selected in the all tab
image

Other than this, lgtm so approving in advance.

**Comment by KaleemNeslit** [comment]
@mzparacha fixed mentioned comments also showed All/Overview filter on all other sorts filters as well.
@Israellund Kindly review it again Fixed mentioned refactors.

**Comment by Israellund** [comment]
@KaleemNeslit This looks good, feel free to merge today. thanks

**Pull request merged**